lib/spack/spack/package.py

```diff
@@ -1286,6 +1297,11 @@ def do_install(self,
                     **kwargs
                 )
 
+        if (kwargs.get('use_cache', False) and
+                self.try_install_from_binary_cache(explicit)):
+            tty.msg('Installed %s from binary cache' % self.name)
+            return
```

Comment:
Should this mirror the install output more closely? (For example, give the path of the installed binary)